File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.35.5.c
/* { dg-do compile } */

void work (int, int);

void
wrong5 (int n)
{
#pragma omp parallel
  {
#pragma omp critical
    {
      work (n, 0);
/* incorrect nesting of barrier region in a critical region */
#pragma omp barrier	/* { dg-error "may not be closely nested" } */
      work (n, 1);
    }
  }
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.35.5.f90
! { dg-do compile }

      SUBROUTINE WRONG5(N)
      INTEGER N
!$OMP PARALLEL DEFAULT(SHARED)
!$OMP CRITICAL
            CALL WORK(N,1)
! incorrect nesting of barrier region in a critical region
!$OMP BARRIER	! { dg-error "region may not be closely nested inside of" }
            CALL WORK(N,2)
!$OMP END CRITICAL
!$OMP END PARALLEL
      END SUBROUTINE WRONG5
